### ISSUE TITLE
GraphError should extend Error

### ISSUE THREAD

**Issue by OlivierCuyp** [microsoftgraph/msgraph-sdk-javascript#248]
Feature Request
Problem
Using this library in NodeJS Backend, most logger detect Error objects and parse them accordingly.
In addition, GraphError not being extended from Error doesn't have a stack trace.
GraphError is not parse correctly by our logger (pinoJS) and we don't have any stack trace.
Solution
Extend GraphError from base JS Error.
I believe it should be done here:

  
    
      msgraph-sdk-javascript/src/GraphError.ts
    
    
         Line 20
      in
      7529db8
    
    
    
    

        
          
           export class GraphError {
[AB#5832](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/5832)

**Comment by naugtur** [comment]
The missing stacktrace is a serious issue. It's not practical to build a product with a team if stacktraces are missing.  

How can I help? Is this a thing someone proficient with JS but not TypeScript can reasonably contribute?

 a workaround you can use in Node to provide a stacktrace:
```js
function track() {
  const st = {};
  Error.captureStackTrace(st);
  return (e) => {
    e.trace = st.stack;
    throw e;
  };
}

await client.api(`/me`).get().catch(track());
```

With a bit of work you can also construct a working Error type object. Capturing the stack trace is the important part.
I wonder if it'd be possible to monkey-patch GraphError from the outside. 

@OlivierCuyp Did you figure something out?

**Comment by OlivierCuyp** [comment]
@nikithauc I made a quick PR let me know if it is ok for you.
I believe the code is retro-compatible with the exception of the default GraphError message which is `""` instead of `null` now.

**Comment by nikithauc** [comment]
@OlivierCuyp Thank you so much for this! I will take a look the pr shortly.

**Comment by naugtur** [comment]
How do I check to see when it gets released?

**Comment by nikithauc** [comment]
This was released in 2.1.0 version of Microsoft Graph Client. 
Release notes mentions the PR for this issue. 

I will work on tagging the issues with the release version.
